src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -97,39 +98,56 @@ private LanguageExpression ConvertGetSecretFunctionCall(InstanceFunctionCallSynt
                     )
             );
         }
+
+        public LanguageExpression ConvertExpression(SyntaxBase syntax)
+            => ConvertExpression(ExpressionBuilder.Build(syntax));
+
         /// <summary>
         /// Converts the specified bicep expression tree into an ARM template expression tree.
         /// The returned tree may be rooted at either a function expression or jtoken expression.
         /// </summary>
         /// <param name="expression">The expression</param>
-        public LanguageExpression ConvertExpression(SyntaxBase expression)
+        public LanguageExpression ConvertExpression(Expression expression)
         {
+            SyntaxBase syntax;
             switch (expression)
             {
-                case BooleanLiteralSyntax boolSyntax:
-                    return CreateFunction(boolSyntax.Value ? "true" : "false");
+                case BooleanLiteralExpression @bool:
+                    return CreateFunction(@bool.Value ? "true" : "false");
 
-                case IntegerLiteralSyntax integerSyntax:
-                    return ConvertInteger(integerSyntax, false);
+                case IntegerLiteralExpression @int:
+                    return @int.Value switch {
+                        // Bicep permits long values, but ARM's parser only permits int jtoken expressions.
+                        // We can work around this by using the `json()` function to represent non-integer numerics.
+                        > int.MaxValue or < int.MinValue => CreateFunction("json", new JTokenExpression(@int.Value.ToString(CultureInfo.InvariantCulture))),
```

Comment:
I think this has been fixed:
![image](https://user-images.githubusercontent.com/705500/202487939-a5b15a25-905c-4ad7-aff8-fde71d3fcdc5.png)

Reply:
This is specifically just for language expressions - JSON numeric literals don't have the same problem. For example:
`"[sum(0, 2147483647)]"` is perfectly fine, but `"[sum(0, 9223372036854775807)]"` has to be represented as `"[sum(0, json('9223372036854775807'))]"`.

My understanding is that it's purely a problem with the language expression **parser** - it only handles ints. So `json('...')` bypasses the parser and allows direct conversion to `JValue` (which doesn't have the same limitation).

Reply:
TIL. I'm glad we can hide that complexity in templates compiled from Bicep, but that should be fixed in the engine for users who are authoring templates in JSON.